parser/lex.go

```diff
@@ -142,7 +144,7 @@ type stateFn func(*lexer) stateFn
 
 // lexer holds the state of the scanner.
 type lexer struct {
-	source  inputSource  // the name of the input; used only for error reports
+	source  InputSource  // the name of the input; used only for error reports
 	input   string       // the string being scanned
 	state   stateFn      // the next lexing function to enter
 	pos     bytePosition // current position in the input
```

Comment:
Copied the style from the golang lexer, so I think it is okay (if not widely used)